2.1 documentation (#26)
Intermodalics/pickit-docs: docs/documentation/teach/explaining-the-teach-detection-parameters.rst

.. _Explaining-the-teach-detection-parameters:

Explaining the Teach detection parameters
-----------------------------------------

The Pickit Teach vision engine is designed to detect complex
3D shapes. This article explains its detection parameters.

.. contents::
    :backlinks: top
    :local:
    :depth: 1

> M-HD preset(M-HD camera only)

Reviewer: Same comment as for other engines, centralize this in a single place.

> M-HD preset(M-HD camera only)

Reviewer: ```suggestion
M-HD preset (M-HD camera only)
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. image:: /assets/images/Documentation/m-hd-preset-21.png

In this tab the preset for the M-HD camera is chosen. This preset determines the settings of the camera and how a point cloud is captured.

This guide helps you chosing a good preset for your application, :ref:`how-to-mhd-preset`.

Define your model(s)
~~~~~~~~~~~~~~~~~~~~

.. image:: /assets/images/Documentation/teach-model-grid-21.png

A list of thaught models is shown accompanied by their ID and
the number of points. By default, there are no models defined.

Add a model
^^^^^^^^^^^

> Press :guilabel:`Add a model` to add your first model, the points

Reviewer: ```suggestion
Press :guilabel:`Add a model` to teach your first model, the points
```

currently visible within the ROI are now saved into a model. When the
model is successfully created, the model will automatically be shown in

> the **Model** view.

Reviewer: ```suggestion
the :ref:`Model <model-view>` view.
```

Label needs to be created in `viewer.rst`

In one product file up to 8 different models can be taught. 
This means that Pickit Teach is capable of looking for 8 different shapes in one detection. 

> See :ref:`how-to-multiple-models` on how you can use the model id in a robot program.

Reviewer: ```suggestion
See :ref:`how-to-multiple-models` on how you can use the model ID in a robot program.
```

Re-teach a model
^^^^^^^^^^^^^^^^

> You can always reteach a model by pressing :guilabel:`Re-teach` for

Reviewer: ```suggestion
You can always re-teach a model by pressing :guilabel:`Re-teach` for
```

a specific model.
By pressing this button, the points currently visible within the ROI
are saved into the model.

.. warning:: When re-teaching a specific model, the point-cloud
   data from the previous model is overwritten. This action cannot be
   undone.

Delete a model
^^^^^^^^^^^^^^

To delete a model, press :guilabel:`Delete` for the specific model.

> The action needs to be confirmed in deletion popup.

Reviewer: ```suggestion
The action needs to be confirmed in a popup.
```

> .. warning:: Model ID's will be reassigned after deleting a certain

Reviewer: ```suggestion
.. warning:: Model ID's will be reassigned after deleting a 
```

>    model. Always check your robot program after deleting models.

Reviewer: ```suggestion
   model. If you use model IDs in your robot program, make sure they're up to date after deleting models.
```
